core/src/main/java/io/grpc/internal/ServerImpl.java

```diff
@@ -808,9 +808,10 @@ void setListener(ServerStreamListener listener) {
     /**
      * Like {@link ServerCall#close(Status, Metadata)}, but thread-safe for internal use.
      */
-    private void internalClose(Throwable t) {
+    private void internalClose(Throwable t, String task) {
       // TODO(ejona86): this is not thread-safe :)
-      stream.close(Status.UNKNOWN.withCause(t), new Metadata());
+      String description = "Internal Application Error @ task " + task;
```

Comment:
Considered ` @ context ` instead of ` @ task`. Let me know if you like one or another better.